### DISCUSSION TITLE
Why "enableGlobalCache" is false by default?

### DISCUSSION THREAD

**Discussion by simnalamburt** [yarnpkg/berry#3462] [Q&A] [answered]
Does anyone know...

1. Why "enableGlobalCache" is false by default?
2. What's the downside of "enableGlobalCache" option being enabled?
3. What's the upside of "enableGlobalCache" option being disabled?

**Comment by saschanaz** [comment]
Probably something to do with #3403?

**Comment by arcanis** [comment, marked as answer]
I think you found [the answer](https://twitter.com/arcanis/status/1333330928953921536), but the gist is that the offline mirror is an old Yarn feature (it was already there in 1.x) that very few people were aware of, despite its benefits. Given that Yarn has a sizeable usage amongst company-grade applications, I felt like enabling it by default would help make our users' deployments generally safer - and it can fairly easily be disabled if not needed.

> What's the downside of "enableGlobalCache" option being enabled?

You rely on the npm registry for your deployments, which is known for being down semi-regularly. Every time I used to notice it was down, it was because people were [pinging the @yarnpkg account](https://twitter.com/yarnpkg/status/1105951094780370945); meanwhile my own builds all passed. It felt like a recurrent enough problem to be worth changing the default.

There are also some fringe edge cases w/ PnP, although they are somewhat unlikely to be triggered in practice.

> What's the upside of "enableGlobalCache" option being disabled?

Your cache is inside your project, so once you clone it you're no longer dependant on the network.

**Reply by simnalamburt** [reply]
Thanks for the answer!

Is there no other way to use global cache without relying on NPM? For example, what will happen if I make hardlinks or symlinks in `./.yarn/cache` from `~/.yarn/berry/cache/` instead of copy? May I ask the reason why relying on global cache makes you rely on NPM?

**Reply by arcanis** [reply]
> what will happen if I make hardlinks or symlinks in `./.yarn/cache` from `~/.yarn/berry/cache/` instead of copy?

While Yarn could create the local cache using symlinks, it would quickly become irrelevant: Git won't create hardlinks on clone, so anyone checking out the repository would use the full size. This, combined with how we can *in theory* benefit from copy-on-write (which is safer than hardlinks, since it doesn't run the risk of accidentally corrupting the global archives), made hardlinks not that interesting.

> May I ask the reason why relying on global cache makes you rely on NPM?

Where would Yarn retrieve the archives from on CI, if not the local cache?

**Reply by simnalamburt** [reply]
Oh, I get it. So to fully utilize the upside of the local cache mode in the company-grade use case, one must be using the Zero-Install workflow. Since with the `yarn install` workflow, one must rely on the NPM registry in the first `yarn install` call which will make the upside of local cache mode irrelevant. Am I right?
